framework/src/modules/chain/logic/init_transaction.js

```diff
@@ -102,7 +102,8 @@ const outTransferAsset = raw => {
 };
 
 class Transaction {
-	constructor(transactions = {}) {
+	constructor({ registeredTransactions = {}, applicationState = {} }) {
+		this.state = applicationState;
```

Comment:
I think applicationState, which are created by the module itself shouldn't be used in the module itself.
it should be using the state that the module has (which should be always correct)
It will create the complexity in the timing management and race condition

Reply:
Ok. Using `lastBlock` now but I had to use `bindModules` inside initTransaction =\